client/jetpack-cloud/sections/agency-dashboard/sites-overview/utils.jsx

```diff
@@ -0,0 +1,308 @@
+import { Gridicon } from '@automattic/components';
+import { translate } from 'i18n-calypso';
+import { useRef, useState } from 'react';
+import Badge from 'calypso/components/badge';
+import Tooltip from 'calypso/components/tooltip';
+
+/**
+ * Returns link and tooltip for each feature based on status
+ * which will be used to format row values. link will be used
+ * to redirect the user when clicked on the row and tooltip is
+ * used to show the tooltip when hovered over the row
+ *
+ * @param  {type} type  Feature type
+ * @param  {status} status Feature status
+ * @param  {siteUrl} siteUrl Site URL
+ * @param  {siteId} siteId Site ID
+ * @returns {object} link & tooltip
+ */
+const getLinks = ( type, status, siteUrl, siteId ) => {
+	let link = '';
+	let tooltip = '';
+	switch ( type ) {
+		case 'backup': {
+			if ( status === 'inactive' ) {
+				link = `/partner-portal/issue-license/?site_id=${ siteId }`;
+			} else {
+				link = `/backup/${ siteUrl }`;
+			}
+			if ( status === 'progress' ) {
+				tooltip = translate( 'Backup in progress' );
+			}
+			break;
+		}
+		case 'scan': {
+			if ( status === 'inactive' ) {
+				link = `/partner-portal/issue-license/?site_id=${ siteId }`;
+			} else {
+				link = `/scan/${ siteUrl }`;
+			}
+			if ( status === 'progress' ) {
+				tooltip = translate( 'Scan in progress' );
+			}
+			break;
+		}
+		case 'monitor': {
+			if ( status === 'failed' ) {
+				link = `https://jptools.wordpress.com/debug/?url=${ siteUrl }`;
+			}
+			if ( status === 'success' ) {
+				tooltip = translate( 'Monitor is on and your site is online' );
+			}
+			break;
+		}
+		case 'plugin': {
+			if ( status === 'warning' ) {
+				link = `https://wordpress.com/plugins/updates/${ siteUrl }`;
+			}
+			break;
+		}
+	}
+	return { link, tooltip };
+};
+
+/**
+ * Returns an object which holds meta data required to format
+ * the row
+ *
+ * @param  {rows} rows  Holds row objects(sites)
+ * @param  {type} type Feature type
+ * @returns {object} Object that holds row(individual row based on type), link(to redirect on click), siteError,
+ * tooltip & tooltip id
+ */
+const getRowMetaData = ( rows, type ) => {
+	const row = rows[ type ];
+	const siteUrl = rows.site?.value?.url;
+	const siteError = rows.site?.error;
+	const siteId = rows.site?.value?.blog_id;
+	const { link, tooltip } = getLinks( type, row.status, siteUrl, siteId );
+	return {
+		row,
+		link,
+		siteError,
+		tooltip,
+		tooltipId: `${ siteId }-${ type }`,
+	};
+};
+
+export default function StatusContent( { content, tooltip, tooltipId, link, siteError } ) {
```

Comment:
Is this intended to be the default export from a utils file? It seems like a component should be in its own file.